### PR TITLE
Make sure the app works for API >= 9

### PR DESCRIPTION
**Pull request by bassosimone**
xref: measurement-kit/measurement-kit#1051

### CODE REVIEW

File: app/src/main/java/org/openobservatory/ooniprobe/data/TestData.java [hunk 1]
```diff
@@ -179,6 +181,15 @@ else if (testName.compareTo(OONITests.WEB_CONNECTIVITY) == 0) {
                         w.set_options("save_real_probe_cc", boolToString(include_cc));
                         w.set_options("no_collector", boolToString(!upload_results));
                         w.set_options("collector_base_url", collector_address);
+                        w.on_progress(new org.openobservatory.measurement_kit.nettests.ProgressCallback() {
+                            @Override
+                            public void callback(double percent, String msg) {
+                                System.out.println("test progress "+ percent);
+                                currentTest.progress = (int)percent*100;
+                                //TODO crases here maybe we need to implement the LocalBroadcastManager?
+                                //TestData.getInstance(activity).notifyObservers();
```
Comment (bassosimone) [inline app/src/main/java/org/openobservatory/ooniprobe/data/TestData.java]: @lorenzoPrimi what kind of crash is this? A segfault or a Java exception? Is there a stacktrace?

Comment (lorenzoPrimi) [inline app/src/main/java/org/openobservatory/ooniprobe/data/TestData.java]: JNI NewStringUTF called with pending exception android.view.ViewRootImpl$CalledFromWrongThreadException

you can try to uncomment the commented code and run

Comment (bassosimone) [inline app/src/main/java/org/openobservatory/ooniprobe/data/TestData.java]: Can you use `runOnUiThread`? (See for example how it was used in measurement-kit/android-example#d69e258a64219cd137bdbc2475b2f4c89f59fbc4)

File: app/src/main/java/org/openobservatory/ooniprobe/data/TestData.java [hunk 2]
```diff
@@ -179,6 +181,15 @@ else if (testName.compareTo(OONITests.WEB_CONNECTIVITY) == 0) {
...
+                            }
+                        });
```
Comment (bassosimone) [inline app/src/main/java/org/openobservatory/ooniprobe/data/TestData.java]: I think the reason why you see the exception is that the callback is called from a background thread and hence you cannot modify the UI from there. Try with `runOnUiThread` as suggested in https://github.com/measurement-kit/ooniprobe-android/pull/22/files#r94856403

**Comment by bassosimone** [comment]
"Working progressbar" <- yeah!!!

**Comment by lorenzoPrimi** [comment]
yes, totally

**Comment by bassosimone** [comment]
Perfect, I'll review through it tomorrow, then!

**Comment by bassosimone** [comment]
Tested on a selection of emulator APIs and ARCHs. Merging!

**Pull request merged**